code/game/objects/structures/false_walls.dm

```diff
@@ -262,22 +262,23 @@
 
 /obj/structure/falsewall/plasma/attackby(obj/item/W, mob/user, params)
 	if(W.is_hot() > 300)
-		var/turf/T = get_turf(src)
-		message_admins("Plasma falsewall ignited by [ADMIN_LOOKUPFLW(user)] in [ADMIN_VERBOSEJMP(T)]")
-		log_game("Plasma falsewall ignited by [key_name(user)] in [AREACOORD(T)]")
-		burnbabyburn()
+		if(plasma_canignite())
+			new /obj/structure/girder/displaced(loc)
+			plasma_ignition(6, user)
 	else
 		return ..()
 
-/obj/structure/falsewall/plasma/proc/burnbabyburn(user)
-	playsound(src, 'sound/items/welder.ogg', 100, 1)
-	atmos_spawn_air("plasma=400;TEMP=1000")
-	new /obj/structure/girder/displaced(loc)
-	qdel(src)
-
 /obj/structure/falsewall/plasma/temperature_expose(datum/gas_mixture/air, exposed_temperature, exposed_volume)
-	if(exposed_temperature > 300)
-		burnbabyburn()
+	if(exposed_temperature > 300 && air.get_moles(GAS_O2) >= PLASMA_MINIMUM_OXYGEN_NEEDED)
+		if(plasma_canignite())
+			new /obj/structure/girder/displaced(loc)
+			plasma_ignition(6)
+
+/obj/structure/falsewall/plasma/bullet_act(obj/item/projectile/Proj)
+	if(!(Proj.nodamage) && Proj.damage_type == BURN)
+		new /obj/structure/girder/displaced(loc)
```

Comment:
this spawns regardless of if it can actually ignite

Reply:
👍 
Moved this under `if(plasma_ignition(6, Proj?.firer))